exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileWrapper.java

```diff
@@ -75,24 +75,31 @@ public ProfileWrapper(final QueryProfile profile) {
     final List<OperatorWrapper> ows = new ArrayList<>();
     // temporary map to store (major_id, operator_id) -> [(op_profile, minor_id)]
     final Map<ImmutablePair<Integer, Integer>, List<ImmutablePair<OperatorProfile, Integer>>> opmap = new HashMap<>();
+    // temporary map to store (major_id, operator_id) -> [minorFragHostname]
```

Comment:
Effectively, what this is doing is converting the opmap from a map of immutable pairs to a map of immutable triples. About here, we might want to restructure the map to point to a class which holds the two elements in the first pair, plus the host name.

Why? Simpler. Easier to keep class fields in sync than to keep two separate lists in sync.

Reply:
+1 on this comment. However, I didn't what to make the PR look to radically alter what is already there. I'll incorporate the change. This should go nicely with a fix for DRILL-5802 and make life a bit easier